internal/operators/manager.go

```diff
@@ -242,6 +243,64 @@ func (mgr *Manager) getEnabledOperators(operators models.Operators) map[models.O
 	return visited
 }
 
+// GetMemoryRequirementForRole returns the amount of usable memory required in Bytes in the host to be able to install all the enabled operators and their dependencies.
+// The value is determined by the sum of each of the enabled operators.
+func (mgr *Manager) GetMemoryRequirementForRole(cluster *common.Cluster, role models.HostRole) (int64, error) {
+	operators, err := unmarshallOperators(cluster.Operators)
+	if err != nil {
+		return 0, err
+	}
+	var t int64
+	for _, o := range operators {
+		switch role {
```

Comment:
The role does not change during the iteration, but I guess the compiler will optimize it not to make same check repeatedly.
WDYT about something like:
- switch on role to decide on which function to use and assign it to a function variable
- iterate over the operators only calling the function in the variable.

Reply:
You're right. I'll rework this out based on your suggestion. Thanks! :smile: